infrastructure/modules/migration/envs/dev/configs/task_settings.json

```diff
@@ -0,0 +1,205 @@
+{
+    "Logging": {
+        "EnableLogging": true,
+        "EnableLogContext": true,
+        "LogComponents": [
+            {
+                "Severity": "LOGGER_SEVERITY_DEFAULT",
+                "Id": "TRANSFORMATION"
+            },
+            {
+                "Severity": "LOGGER_SEVERITY_DEFAULT",
+                "Id": "SOURCE_UNLOAD"
+            },
+            {
+                "Severity": "LOGGER_SEVERITY_DEFAULT",
+                "Id": "IO"
+            },
+            {
+                "Severity": "LOGGER_SEVERITY_DEFAULT",
+                "Id": "TARGET_LOAD"
+            },
+            {
+                "Severity": "LOGGER_SEVERITY_DEFAULT",
+                "Id": "PERFORMANCE"
+            },
+            {
+                "Severity": "LOGGER_SEVERITY_DEFAULT",
+                "Id": "SOURCE_CAPTURE"
+            },
+            {
+                "Severity": "LOGGER_SEVERITY_DEFAULT",
+                "Id": "SORTER"
+            },
+            {
+                "Severity": "LOGGER_SEVERITY_DEFAULT",
+                "Id": "REST_SERVER"
+            },
+            {
+                "Severity": "LOGGER_SEVERITY_DEFAULT",
+                "Id": "VALIDATOR_EXT"
+            },
+            {
+                "Severity": "LOGGER_SEVERITY_DEFAULT",
+                "Id": "TARGET_APPLY"
+            },
+            {
+                "Severity": "LOGGER_SEVERITY_DEFAULT",
+                "Id": "TASK_MANAGER"
+            },
+            {
+                "Severity": "LOGGER_SEVERITY_DEFAULT",
+                "Id": "TABLES_MANAGER"
+            },
+            {
+                "Severity": "LOGGER_SEVERITY_DEFAULT",
+                "Id": "METADATA_MANAGER"
+            },
+            {
+                "Severity": "LOGGER_SEVERITY_DEFAULT",
+                "Id": "FILE_FACTORY"
+            },
+            {
+                "Severity": "LOGGER_SEVERITY_DEFAULT",
+                "Id": "COMMON"
+            },
+            {
+                "Severity": "LOGGER_SEVERITY_DEFAULT",
+                "Id": "ADDONS"
+            },
+            {
+                "Severity": "LOGGER_SEVERITY_DEFAULT",
+                "Id": "DATA_STRUCTURE"
+            },
+            {
+                "Severity": "LOGGER_SEVERITY_DEFAULT",
+                "Id": "COMMUNICATION"
+            },
+            {
+                "Severity": "LOGGER_SEVERITY_DEFAULT",
+                "Id": "FILE_TRANSFER"
+            }
+        ],
+        "LogConfiguration": {
+            "TraceOnErrorMb": 10,
+            "EnableTraceOnError": false
+        },
+        "CloudWatchLogGroup": "dms-serverless-replication-57J6Z4LH2JAUNKC3LS7RUETZUE",
+        "CloudWatchLogStream": "dms-serverless-serv-res-id-1761840201151-lqz"
```

Comment:
As above, we'd probably create log groups / streams in a Terraform context, then pass the name into this JSON config using template interpolation.